pkg/tbtc/maintainer/chain.go

```diff
@@ -2,12 +2,38 @@ package maintainer
 
```

Comment:
Shouldn't the `maintainer` package be moved out of the `tbtc` package to `pkg/maintainer` like we have `pkg/sortition`?

If we decide to leave it as is, shouldn't the `BitcoinDifficultyChain` interface be linked to the `tbtc.Chain` interface just like other contracts?

cc @pdyraga and @lukasz-zimnoch

Reply:
Yep, we should definitely have `pkg/maintainer` from the very beginning. Moreover, `pkg/maintainer` should be standalone, just like `pkg/sortition`, and should not depend on `pkg/tbtc` in any way. That means it should declare it's own `maintainer.Chain` interface that will expose their chain's expectations.

Reply:
On the other hand all the maintainers will perform work that pertains to Tbtc.
@pdyraga [suggested](https://github.com/keep-network/keep-core/pull/3374#discussion_r1025290903) that the `maintainer` functionalities should be inside `/pkg/tbtc/maintainer` and not `/pkg/maintainer`.

Reply:
Yeah, that's right but that depends on how we look at our package structure. Currently, we have two packages representing the standalone applications that can be run as part of the client:
- `pkg/beacon` 
- `pkg/tbtc` 

We also have auxiliary packages that can be used by applications to do their work:
- `pkg/bitcoin`
- `pkg/chain`
- `pkg/net`
- `pkg/sortition`
- `pkg/tecdsa`

In the aforementioned structure, the `pkg/tbtc` package actually represents the code of the tbtc node that performs actions specific to a wallet member. This package does not represent the entire tbtc domain so it should not be considered a bag for everything related to tbtc. 

The typical usage of sub-packages in Go is to place implementation adapters. Examples:
- `pkg/net` contains the network domain and `pkg/net/libp2p` its LibP2P-based implementation
- `pkg/bitcoin` contains a part of the Bitcoin domain and `pkg/bitcoin/electrum` holds an Electrum implementation of the Bitcoin client
- `pkg/tecdsa` contains the tECDSA domain objects, the `pkg/tecdsa/dkg` defines how to generate a key, and the `pkg/tecdsa/signing` implements the signing algorithm
- standard library's `net/http` contains the HTTP-related domain and `net/http/httptrace` is an implementation of tracing

In that context, the tbtc maintainer is not an implementation of the tbtc wallet node. It's a standalone being within the same tbtc domain but with its own bounded context (focused around SPV actions). This is why I think it should be placed into its own `pkg/maintainer` package. Additionally, if we need to add another kind of maintainer, not-related to tbtc, the `pkg/maintainer` will be a good match for it. For example, we could argue if the Bitcoin difficulty maintainer is actually something specific to tbtc. Of course it is used as part of tbtc but this is a broader concept that can be reused by other applications.

An open question is whether `pkg/tbtc` should remain as is or be changed to something specific, reflecting that this code is actually about the tbtc wallet node. However, I have no idea about a good proposition here.

Reply:
I was thinking of `pkg/tbtc/maintainer` as an extension of `tbtc` node: some additional responsibilities added to a regular node and all these responsibilities being specific to `tbtc`. We could argue SPV maintainer is not tbtc-related but wallet lifecycle maintainer is definitely tbtc-related.

I think `pkg/sortition` is a different kind of an animal - this is a utility code used by `pkg/tbtc` and `pkg/beacon` nodes.

If you guys think it makes more sense to have `pkg/maintainer` and this will save us a headache, let's go this way.

Reply:
> If you guys think it makes more sense to have `pkg/maintainer` and this will save us a headache, let's go this way

I think so. This structure will allow us to have independent packages that live their own lives. This will reflect that those two packages operate in different contexts of the tbtc domain (node and maintainer). This structure is also more flexible if we need to add non-tbtc maintainers in the future. Having `pkg/tbtc/maintainer` will tempt us to introduce dependencies on the `pkg/tbtc` package to reuse some code from it. However, as mentioned, the `pkg/tbtc` code is tailored for the tbtc node domain context and we can have a hard way to satisfy the maintainer context as well. That would lead to increased complexity and worse code.

Reply:
Moved the maintainer package to `pkg/maintainer` in 1674a6d736b31644dee86192ee653b442f89dd27.

Reply:
Leaving it unresolved for future reference.